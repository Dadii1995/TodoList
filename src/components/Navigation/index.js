import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { NavLink as RRNavLink } from 'react-router-dom'
import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink,
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
  Button,
} from 'reactstrap'
import withTheme from '../../hocs/withTheme'
import withAuth from '../../hocs/withAuth'

class Navigation extends Component {
  constructor(props) {
    super(props)

    this.toggle = this.toggle.bind(this)
    this.state = {
      isOpen: false,
    }
  }
  toggle() {
    this.setState({
      isOpen: !this.state.isOpen,
    })
  }
  render() {
    return (
      <div>
        <Navbar expand="md" style={{ background: this.props.colors.background }}>
          <NavbarBrand href="/">Todo list</NavbarBrand>

          <NavbarToggler onClick={this.toggle} />
          <Collapse isOpen={this.state.isOpen} navbar>
            <Nav className="ml-auto" navbar>
              <NavItem>
                <NavLink
                  activeClassName="active-link"
                  data-cy="nav-home"
                  exact
                  tag={RRNavLink}
                  to="/"
                >
                  Home
                </NavLink>
              </NavItem>
              <UncontrolledDropdown active={true} inNavbar nav>
                <DropdownToggle caret data-cy="nav-profile" nav>
                  Profile
                </DropdownToggle>
                <DropdownMenu right>
                  <DropdownItem>
                    <NavLink
                      activeClassName="active-link"
                      data-cy="nav-profile-details"
                      exact
                      tag={RRNavLink}
                      to="/profile"
                    >
                      Details
                    </NavLink>
                  </DropdownItem>
                  <DropdownItem>
                    <NavLink
                      activeClassName="active-link"
                      data-cy="nav-profile-edit"
                      tag={RRNavLink}
                      to="/profile/edit"
                    >
                      Edit
                    </NavLink>
                  </DropdownItem>
                  <DropdownItem divider />
                  <DropdownItem>
                    <NavLink
                      activeClassName="active-link"
                      data-cy="nav-profile-change-password"
                      tag={RRNavLink}
                      to="/profile/changepassword"
                    >
                      Change Password
                    </NavLink>
                  </DropdownItem>
                </DropdownMenu>
              </UncontrolledDropdown>
              <NavItem>
                <NavLink
                  activeClassName="active-link"
                  data-cy="nav-blog"
                  tag={RRNavLink}
                  to="/blog"
                >
                  Blog
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink
                  activeClassName="active-link"
                  data-cy="nav-weather"
                  tag={RRNavLink}
                  to="/weather"
                >
                  Weather
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink
                  activeClassName="active-link"
                  data-cy="nav-contact"
                  tag={RRNavLink}
                  to="/contact"
                >
                  Contact
                </NavLink>
              </NavItem>
              {this.props.isAuth ? (
                <Button
                  color="warning"
                  data-cy="nav-loguot-button"
                  onClick={() => this.props.logOut()}
                >
                  Log out
                </Button>
              ) : (
                <NavItem>
                  <NavLink activeClassName="active-link" tag={RRNavLink} to="/login">
                    Log in
                  </NavLink>
                </NavItem>
              )}
            </Nav>
          </Collapse>
        </Navbar>
      </div>
    )
  }
}

Navigation.propTypes = {
  colors: PropTypes.shape({
    background: PropTypes.string.isRequired,
    primary: PropTypes.string,
  }),
}

export default withAuth(withTheme(Navigation))
