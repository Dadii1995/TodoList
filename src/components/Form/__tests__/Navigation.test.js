import renderer from 'react-test-renderer'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'

import Navigation from '../../Navigation'

const getWrapper = () => {
  const wrapper = renderer.create(
    <Router>
      <Navigation />
    </Router>,
  )
  const root = wrapper.root
  return { root, wrapper }
}
describe('<Navigation/>', () => {
  test('renders', () => {
    const { wrapper } = getWrapper()
    expect(wrapper).toMatchSnapshot()
  })
})
