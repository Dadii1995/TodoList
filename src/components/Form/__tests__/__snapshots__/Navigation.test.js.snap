// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation/> renders 1`] = `
<div>
  <nav
    className="navbar navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      reactstrap
    </a>
    <ul
      className="nav"
    >
      <li
        className="nav-item"
      >
        <a
          aria-current="page"
          className="nav-link active-link"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </li>
      <li
        className="dropdown nav-item"
        onKeyDown={[Function]}
      >
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle nav-link"
          href="#"
          onClick={[Function]}
        >
          Profile
        </a>
        <div
          aria-hidden={true}
          className="dropdown-menu dropdown-menu-right"
          role="menu"
          tabIndex="-1"
        >
          <button
            className="dropdown-item"
            onClick={[Function]}
            role="menuitem"
            tabIndex="0"
            type="button"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/profile"
              onClick={[Function]}
            >
              Details
            </a>
          </button>
          <button
            className="dropdown-item"
            onClick={[Function]}
            role="menuitem"
            tabIndex="0"
            type="button"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/profile/edit"
              onClick={[Function]}
            >
              Edit
            </a>
          </button>
          <div
            className="dropdown-divider"
            onClick={[Function]}
            tabIndex="-1"
          />
          <button
            className="dropdown-item"
            onClick={[Function]}
            role="menuitem"
            tabIndex="0"
            type="button"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/profile/changepassword"
              onClick={[Function]}
            >
              Change Password
            </a>
          </button>
        </div>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/blog"
          onClick={[Function]}
        >
          Blog
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/weather"
          onClick={[Function]}
        >
          Weather
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/contact"
          onClick={[Function]}
        >
          Contact
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
